### DISCUSSION TITLE
Real Hardware Booting

### DISCUSSION THREAD

**Discussion by atlassoftware-cpu** [CosmosOS/Cosmos#2986] [Q&A]
So, i made an os, its great, has mouse support. but i tried changing it to ISO Image and burning it using Rufus as the website said. then booted it. it showed the Welcome to Grub text. then showed blackscreen (it should show the background which is bright not black so its noticable) please help. thanks

**Comment by ghost** [comment]
is there a VFS?

**Reply by ghost** [reply]
> is there a VFS?

I Think It Can Have VFS In This Video He Has A OS Running In This Video With VFS: https://www.youtube.com/watch?v=l-ZGWNC3ioo

**Comment by TheVogels** [comment]
Cosmos does not seem to work on UEFI which is on most new devices, I have only got to that screen too.
I have however ran it on a DELL Inspiron laptop that uses BIOS through the DVD RW/CD RW drive with a DVD RW and it ran.
On my normal UEFI laptop I noticed it does not let me manually specify my USB DVD RW/CD RW drive and fails to boot in it with the legacy disk drive option, I put it on a USB drive using Rufus and it only got to Welcome to Grub on the legacy option and gives a black screen when booting to the manually specified file, which is OS.bin.

**Reply by TheVogels** [reply]
I just tried running the USB drive on the Inspiron laptop and it said Welcome to Grub and then went to Cosmos.

**Reply by ghost** [reply]
Does it run with Cosmos with VFS? And which laptop model do you use? Did you format the drive of the laptop in FAT32?
Best regards

**Reply by TheVogels** [reply]
> Does it run with Cosmos with VFS? And which laptop model do you use? Did you format the drive of the laptop in FAT32? Best regards

If I remember correctly, the only drive was the USB Stick and when I told it to register VFS it would crash.

**Comment by tails1154** [comment]
Try using a disc. It works for me. USB Flash drives also don't even boot for me

**Comment by Guillermo-Santos** [comment]
Also remember to disable debugging

**Comment by randomusert** [comment]
I have HP laptops and they also refuse to Start The oseven after changing BIOS settings
